src/components/utils/test/thread_test.cc

```diff
@@ -74,14 +78,18 @@ class ThreadTestDelegate : public threads::ThreadDelegate {
   void WaitForThreadStart() {
     AutoLock test_lock(test_mutex_);
     while (!thread_started_) {
-      cond_var_.Wait(test_lock);
+      cond_var_.WaitFor(test_lock, kMilliSecondsToWait);
```

Comment:
@wolfylambova22 what will happen in case of `timeout`?
I beleive we have to stop the test isn't it? In your test case you will continue to wait again.
And also you can remove the while loop at all. `WaitFor` absolutely substutute this one